src/systemFlake.nix

```diff
@@ -189,21 +198,11 @@ mergeAny otherArguments (
 
         pkgs = mapAttrs importChannel channels;
 
-        mkOutput = output: builder:
-          mergeAny
-            # prevent override of nested outputs in otherArguments
-            (optionalAttrs (otherArguments ? ${output}.${system})
-              { ${output} = otherArguments.${output}.${system}; })
-            (optionalAttrs (args ? ${builder})
-              { ${output} = args.${builder} pkgs; });
+        outputs = mapAttrs
+          (outputName: output: otherArguments.${outputName}.${system} or { } // output)
+          (outputsBuilder pkgs);
```

Comment:
I've tried filtering for empty attrsets or `null` here, but that can cause infinite recursion if any output depends on another.

Reply:
Maybe we can be a little more restrictive and require users to set `default*` in terms of `self.packages.${system}.mypackage` instead of provide the possibility of a full blown builder, that receives channels attribute. (Same for apps.)

What do you think?

Reply:
That is generally a good practice. I have no idea how to implement a restriction for that. If we try and check if something is referring to `self` we would probably run into infinite recursion.